src/phoenix/experimental/evals/long_context_evaluators.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Optional
+
+from phoenix.experimental.evals import PromptTemplate
+from phoenix.experimental.evals.models import BaseEvalModel
+
+
+class Refiner:
+    def __init__(
+        self,
+        model: BaseEvalModel,
+        initial_prompt_template: PromptTemplate,
+        refine_prompt_template: PromptTemplate,
+        reduce_prompt_template: Optional[PromptTemplate] = None,
+    ) -> None:
+        self._model = model
+        self._initial_prompt_template = initial_prompt_template
+        self._refine_prompt_template = refine_prompt_template
+        self._reduce_prompt_template = reduce_prompt_template
+
+    def evaluate(self, chunks: List[str]) -> str:
+        if len(chunks) < 2:
+            raise ValueError("Must provide two or more documents")
+        model = self._model
+        initial_prompt = self._initial_prompt_template.format({"chunk": chunks[0]})
+        accumulator = model(initial_prompt)
```

Comment:
I think this might be a tad over confident that the model invocation returns data in a way that's accumulated. But honestly maybe this is actually the simplest approach. Was that your thought as well?

Reply:
I think it depends heavily on the initial prompt. In the notebook, we start out with an evaluation of the summary relative to the first context (including an explanation), and we iteratively refine that evaluation and explanation. This seems to be the strategy taken by LangChain + LlamaIndex as far as I can tell. Importantly, it is not a single classification ("good" or "bad") that is accumulated in the accumulator, but a concise explanation that is iteratively refined.